AWSTranscribeStreamingUnitTests/AWSTranscribeStreamingClientWebSocketProviderTests.swift

```diff
@@ -0,0 +1,332 @@
+            return
+        }
+        
+        request.languageCode = .enUS
+        request.mediaEncoding = .pcm
+        request.mediaSampleRateHertz = 8000
+        
+        let delegate = MockTranscribeStreamingClientDelegate()
+        
+        let receiveClosed = expectation(description: "Callback received closed status")
+        delegate.connectionStatusCallback = { status, error in
+            if status == AWSTranscribeStreamingClientConnectionStatus.closed {
+                receiveClosed.fulfill()
+            }
+        }
+        
+        client.setDelegate(delegate, callbackQueue: DispatchQueue.global())
+        
+        client.startTranscriptionWSS(request)
+        mockWebSocketProvider.disconnect()
+        wait(for: [receiveClosed], timeout: 10)
+    }
+    
+    // Given: A web socket provider
+    // When: when registering the streaming client
+    // Then: the streaming client is able to use a third party web socket provider and open the socket and send data
+    func testWSSSendData() {
+        let mockWebSocketProvider = MockWebSocketProvider()
+        mockWebSocketProvider.setErrorOnConnectionStatusDidChange(error: nil)
+        guard let config = AWSServiceConfiguration(
+            region: .USEast1,
+            credentialsProvider: MockReturningCredentialsProvider()) else {
+                XCTFail("Couldn't initialize config with MockErrorReturningCredentialsProvider")
+                return
+        }
+        
+        AWSTranscribeStreaming.register(with: config, forKey: "testWSSInitWithWebSocketProvider", webSocketProvider: mockWebSocketProvider)
+        
+        let client = AWSTranscribeStreaming(forKey: "testWSSInitWithWebSocketProvider")
+        
+        guard let request = AWSTranscribeStreamingStartStreamTranscriptionRequest() else {
+            XCTFail("request unexpectedly nil")
+            return
+        }
+        let transcription = "some transcription"
+        let data = Data(transcription.utf8)
+        
+        request.languageCode = .enUS
+        request.mediaEncoding = .pcm
+        request.mediaSampleRateHertz = 8000
+        
+        let delegate = MockTranscribeStreamingClientDelegate()
+        
+        let receiveDataSend = expectation(description: "Callback received data")
+        delegate.connectionStatusCallback = { status, error in
+            
+            if status == AWSTranscribeStreamingClientConnectionStatus.connected {
+                //send data once socket is connected
+                
+                mockWebSocketProvider.send(data)
+            }
+        }
+        delegate.receiveEventCallback = { data, error in
+            if error == nil {
+                receiveDataSend.fulfill()
+            }
+        }
+        
+        client.setDelegate(delegate, callbackQueue: DispatchQueue.global())
+        
+        client.startTranscriptionWSS(request)
+        //add waiter to open web socket and make sure its connected before sending data.
```

Comment:
Isn't this handled by L186 above?

Reply:
hm not sure what you mean? this call basically instantiates the web socket and opens its which has to happen before data is sent. line 186 just checks if it's open before sending data.